politeiad/backendv2/tstorebe/plugins/pi/cmds.go

```diff
@@ -242,25 +242,96 @@ func (p *piPlugin) cmdBillingStatusChanges(token []byte) (string, error) {
 	return string(reply), nil
 }
 
+// isFinalStatus determines whether the given proposal status is final and
+// not expected to change in the future.
+func isFinalStatus(status pi.PropStatusT) bool {
+	switch status {
+	case pi.PropStatusUnvettedAbandoned, pi.PropStatusUnvettedCensored,
+		pi.PropStatusAbandoned, pi.PropStatusCensored, pi.PropStatusApproved,
+		pi.PropStatusRejected:
+		return true
+	default:
+		return false
+	}
+}
+
+// needsBillingStatusChanges returns true if the given proposal status
+// is associated with approved proposal which needs only the latest billing
+// status metadata to detemine the proposal proposal status on runtime.
+func needsOnlyBillingStatusChanges(status pi.PropStatusT) bool {
+	switch status {
+	case pi.PropStatusActive, pi.PropStatusCompleted, pi.PropStatusClosed:
+		return true
+	default:
+		return false
+	}
+}
+
 // cmdSummary returns the pi summary of a proposal.
 func (p *piPlugin) cmdSummary(token []byte) (string, error) {
```

Comment:
Unit tests can also now be added to `cmdSummary` when the proposal status is final and cached since it doesn't require any backend calls.  You would pre-load the proposal status cache with the entries.